frameworks/Elixir/phoenix/lib/hello_web/controllers/page_controller.ex

```diff
@@ -29,7 +30,7 @@ defmodule HelloWeb.PageController do
     worlds =
       Stream.repeatedly(&random_id/0)
       |> Stream.uniq()
-      |> Stream.map(fn idx -> Repo.get(World, idx) end)
+      |> Stream.map(&Repo.get(World, &1))
       |> Enum.take(size(params["queries"]))
```

Comment:
We can optimize this in two ways:

1. Wrap the whole operation on a `Repo.checkout`, so we avoid checking out the connection multiple times
2. Use Task.async/await or Task.async_stream so we run the queries in parallel